cmd/ncproxy/ncproxy.go

```diff
@@ -559,6 +559,42 @@ type ttrpcService struct {
 	m sync.Mutex
 }
 
+func getComputeAgentClient(agentAddr string) (computeagent.ComputeAgentService, error) {
+	conn, err := winio.DialPipe(agentAddr, nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to connect to compute agent service")
+	}
+	client := ttrpc.NewClient(
+		conn,
+		ttrpc.WithUnaryClientInterceptor(octtrpc.ClientInterceptor()),
+		ttrpc.WithOnClose(func() { conn.Close() }),
+	)
+	return computeagent.NewComputeAgentClient(client), nil
+}
+
+func (s *ttrpcService) reconnectComputeAgents(ctx context.Context) {
+	contents, err := agentStore.getKeyValueMap(ctx)
+	if err != nil {
+		log.G(ctx).WithError(err).Warn("failed to get compute agent information")
+	}
+	s.m.Lock()
+	defer s.m.Unlock()
+
+	for k, v := range contents {
+		service, err := getComputeAgentClient(v)
+		if err != nil {
+			// can't connect to compute agent, remove entry in database
+			log.G(ctx).WithError(err).Warnf("failed to create new compute agent client for %s", service)
```

Comment:
If the dial fails, won't service be nil? I'd log the address that we tried to dial instead.